src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs

```diff
@@ -175,6 +176,20 @@ public void Configure(IApplicationBuilder app, IOptionsMonitor<TestWebServerOpti
                     });
                 });
             }
+            if (options.WebServerUploadResults)
+            {
+                app.UseRouter(router =>
+                {
+                    router.MapPost("/test-results", async context =>
+                    {
+                        var xmlResultsFilePath = Path.Combine(options.OutputDirectory ?? Directory.CreateTempSubdirectory().FullName, "testResults.xml");
+                        using var fileStream = new FileStream(xmlResultsFilePath, FileMode.Create);
+                        await context.Request.Body.CopyToAsync(fileStream);
+                        await fileStream.FlushAsync();

```

Comment:
nit: [docs](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-9.0#read-the-request-body) suggest this:
```suggestion
                        await using var fileStream = File.Create(xmlResultsFilePath);
                        await request.BodyReader.CopyToAsync(fileStream);
```